builder/frameworks/component_manager.py

```
@@ -0,0 +1,183 @@
+flag_custom_component_add = False
+flag_custom_component_remove = False
+removed_components = set()
+
+ARDUINO_FRAMEWORK_DIR = platform.get_package_dir("framework-arduinoespressif32")
+arduino_libs_mcu = join(ARDUINO_FRAMEWORK_DIR,"tools","esp32-arduino-libs",mcu)
+
+
+def HandleCOMPONENTsettings(env, flag_custom_component_add, flag_custom_component_remove):
+    global removed_components
+    
+    if flag_custom_component_add == True or flag_custom_component_remove == True: # todo remove duplicated
```

Ruff (0.11.9)

Avoid equality comparisons to `True`; use `if flag_custom_component_add:` for truth checks

Replace with `flag_custom_component_add`

(E712)